### PR TITLE
[MAINTENANCE] Refactor `EmailAction` for V1

### PR DESCRIPTION
**Pull request by cdkini**
Update action and renderer to use checkpoint result

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/render/test_EmailRenderer.py [hunk 1]
```diff
@@ -130,3 +130,26 @@ def test_EmailRenderer_get_report_element():
 
     # this should work
     assert email_renderer._get_report_element(docs_link="i_should_work") is not None
+
+
+@pytest.mark.unit
```
Comment (cdkini) [inline tests/render/test_EmailRenderer.py:135]: Screenshot 2024-04-09 at 8 06 49 AM

File: tests/actions/test_core_actions.py [hunk 2]
```diff
@@ -868,14 +868,29 @@ def test_APINotificationAction_run(self, checkpoint_result: CheckpointResult):
...
     @pytest.mark.unit
-    @pytest.mark.xfail(
-        reason="Not yet implemented for this class", strict=True, raises=NotImplementedError
-    )
     def test_EmailAction_run(self, checkpoint_result: CheckpointResult):
         action = EmailAction(
-            smtp_address="test", smtp_port=587, receiver_emails="test1@gmail.com, test2@hotmail.com"
+            smtp_address="test",
+            smtp_port="587",
+            receiver_emails="test1@gmail.com, test2@hotmail.com",
```
Comment (tyler-hoffman) [inline tests/actions/test_core_actions.py]: The implementation looks good, but I'd consider parameterizing the test to prove this works when there's no space, and when there's only one email.

File: tests/actions/test_core_actions.py [hunk 3]
```diff
@@ -868,14 +868,29 @@ def test_APINotificationAction_run(self, checkpoint_result: CheckpointResult):
...
         )
-        action.v1_run(checkpoint_result=checkpoint_result)
+
+        with mock.patch("great_expectations.checkpoint.actions.send_email") as mock_send_email:
+            out = action.v1_run(checkpoint_result=checkpoint_result)
+
+        mock_send_email.assert_called_once_with(
+            title=mock.ANY,
```
Comment (tyler-hoffman) [inline tests/actions/test_core_actions.py:914]: I think ANY makes sense here, but I might add another assertion below to verify that the argument passed kind of looks like the right string. (e.g. `suite_name in call_args.kwargs["title"]` (or whatever the syntax is for that))

**Review by tyler-hoffman — APPROVED** [review]
LGTM

**Pull request merged**